### PR TITLE
Keep requires directive in gateway schema

### LINKED ISSUE #729: @requires directive not found in gateway list of directives
Hi! 

My team writes the gateway schema to disk for a number of reasons. We like seeing what our federated graph schema compiles to and its nice for communicating what the schema looks like to other teams (we dont use a schema registry).

We recently started to use the `@requires` directive in a service and the gateway doesn't list it in the directives available. Because of this, `@requires` does not get defined at the top of the schema file even though its being used in the schema. This breaks our linting and also doesn't follow the GraphQL schema spec closely.

Functionally `@requires` works as expected for the federated services.

## Steps to reproduce

1. Create a new node project on your computer.

1. [Copy and paste the example gateway code](https://github.com/mercurius-js/mercurius/blob/master/examples/gateway.js) from the repo into a index.js file.

1. Run the gateway and the two services (does this by default) by running `node index.js`.

1. Query the gateway for directives

1. See that no requires directive will exist

1. Query service 2 for directives

1. See that the `@requires` directive exists

Here is the query you can run to get directives.
```graphql
query IntrospectionQuery {
  __schema {
    directives {
      name
      description
      locations
    }
  }
}
```

### PR DESCRIPTION
**Pull request by aleccool213**
Fixes #729

### CODE REVIEW

File: test/gateway/pollingInterval.js [hunk 1]
```diff
@@ -710,16 +710,6 @@ test('Polling schemas (should properly regenerate the schema when a downstream s
   })
   t.teardown(() => clock.uninstall())
   const oldSchema = `
-    directive @extends on INTERFACE | OBJECT
```
Comment (aleccool213) [inline test/gateway/pollingInterval.js:713]: It doesn't make sense for a service to have this defined in its schema. This is federation metadata.

File: test/gateway/pollingInterval.js [hunk 2]
```diff
@@ -804,16 +794,6 @@ test('Polling schemas (should properly regenerate the schema when a downstream s
   })
 
   const refreshedSchema = `
-    directive @extends on INTERFACE | OBJECT
```
Comment (aleccool213) [inline test/gateway/pollingInterval.js:807]: It doesn't make sense for a service to have this defined in its schema. This is federation metadata.

**Review by mcollina — APPROVED** [review]
lgtm, good spot!

**Comment by mcollina** [comment]
@jonnydgreen could you take a look as well?

**Comment by jonnydgreen** [comment]
Since it's a bugfix, I'm thinking we should backport this to v8 as well - what do you think @mcollina?

**Comment by mcollina** [comment]
Great idea, let's try the action.

**Pull request merged**